src/components/LeftContainer.js

```diff
@@ -12,41 +12,55 @@ import 'codemirror/theme/dracula.css';
 import 'codemirror/mode/clike/clike';
 import 'codemirror/mode/python/python';
 import 'codemirror/mode/javascript/javascript';
+import 'codemirror/addon/edit/closebrackets';
 import { useTheme } from '../context/Providers/Themeprovider';
+import { useBoilerplate } from '../context/Providers/BoilerplateProvider';
 
 const LeftContainer = ({
   pre, ext, updateOutput, updateLoading,
 }) => {
   const [copied, setCopied] = useState(false);
-  const [code, setCode] = useState(pre);
+  const [code, setCode] = useState();
   const [input, setInput] = useState(null);
   const [fileinput, setFileInput] = useState();
   const [mode, setMode] = useState(ext);
   const { theme } = useTheme();
+  const { boilerplateCode } = useBoilerplate();
 
   useEffect(() => {
     setTimeout(() => {
       setCode(pre);
     }, 0.5);
   }, [pre]);
 
+  useEffect(() => {
+    if (localStorage.getItem(mode)) {
```

Comment:
Please move this to a variable something like:

const cachedUserCode = localStorage.getItem(mode)